tests/std/tests/GH_003663_cast_contiguous_iterator_to_correct_type/test.compile.pass.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <iterator>
+
+class contiguous_iterator {

```

Comment:
Also, it would be good to `static_assert` that this satisfies the `contiguous_iterator` concept, verifying that we've remembered to provide all of the necessary operations (and will therefore activate the optimization whose code needed to be fixed).